tests/test_basic_functionality.py

```diff
@@ -458,6 +458,18 @@ def test_to_dict_yaml(test_spec: Path) -> None:
     assert yaml_config == lct.to_dict()
 
 
+def test_getter() -> None:
+    getter_dict: NestedDict = {
+        "outer_layer_1": "test_value",
+        "outer_layer_2": {"inner_layer": "test_value2"},
+    }
+    lct = LayeredConfigTree(getter_dict)
+
+    assert lct.get("outer_layer_1") == "test_value"
+    assert lct.get("outer_layer_2").to_dict() == getter_dict["outer_layer_2"]  # type: ignore [union-attr]
```

Comment:
mypy is complaining that some of the types returned by get (str, int, list, etc) do not have to_dict or get methods

Reply:
same deal - I think you need to assign and type hint first, e.g. `expected: dict[str, str] = getter_dict["outer_layer_2"]`